File: app/src/main/java/vasuki/istanpdf/model/PageItem.java
package vasuki.istanpdf.model;

import android.graphics.Bitmap;

public class PageItem {
    public final int originalIndex;
    public Bitmap thumbnail;
    public String displayName = "";
    public boolean keep = true;
    public int rotation = 0;

    public PageItem(int originalIndex, Bitmap thumbnail) {
        this.originalIndex = originalIndex;
        this.thumbnail = thumbnail;
    }

    public PageItem(int originalIndex, Bitmap thumbnail, String displayName) {
        this.originalIndex = originalIndex;
        this.thumbnail = thumbnail;
        this.displayName = displayName;
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/model/PageItem.kt
package vasuki.istanpdf.model

import android.net.Uri
import java.io.File

class PageItem(
    val originalIndex: Int,
    var sourceUri: Uri? = null,
    var renderIndex: Int = originalIndex,
    var displayName: String = "",
    var keep: Boolean = true,
    var rotation: Int = 0,
    var replacementFile: File? = null,
    var uri: Uri? = null
) {
    val cacheKey: String get() {
        val baseId = replacementFile?.name ?: sourceUri?.toString() ?: uri?.toString() ?: "none"
        return "${baseId}_${renderIndex}_r${(rotation % 360 + 360) % 360}"
    }
}